torchgeo/datasets/utils.py

```diff
@@ -370,23 +423,83 @@ def working_dir(dirname: str, create: bool = False) -> Iterator[None]:
         os.chdir(cwd)
 
 
-def collate_dict(samples: List[Dict[str, Any]]) -> Dict[str, Any]:
-    """Merge a list of samples to form a mini-batch of Tensors.
+def _list_dict_to_dict_list(samples: Iterable[Dict[Any, Any]]) -> Dict[Any, List[Any]]:
+    """Convert a list of dictionaries to a dictionary of lists.
+
+    Args:
+        samples: a list of dictionaries
+
+    Returns:
+        a dictionary of lists
+    """
+    collated = collections.defaultdict(list)
+    for sample in samples:
+        for key, value in sample.items():
+            collated[key].append(value)
+    return collated
+
+
+def stack_samples(samples: Iterable[Dict[Any, Any]]) -> Dict[Any, Any]:
```

Comment:
For `stack_samples` and `concat_samples`, I chose the names because that's what the PyTorch functions are called. For `merge_samples`, I chose the name because that's what rasterio calls it. I'm open to suggestions for alternate names if these aren't clear.